examples/MomentoLoadGen/Program.cs

```diff
@@ -0,0 +1,394 @@
+
+            GlobalRequestCount = 0;
+            GlobalSuccessCount = 0;
+            GlobalDeadlineExceededCount = 0;
+            GlobalResourceExhaustedCount = 0;
+            GlobalRstStreamCount = 0;
+            GlobalUnavailableCount = 0;
+        }
+    };
+
+
+    public class CsharpLoadGenerator
+    {
+        const int CACHE_ITEM_TTL_SECONDS = 60;
+        const string CACHE_NAME = "momento-loadgen";
+        const int NUM_REQUESTS_PER_OPERATION = 2;
+        const int PRINT_STATS_EVERY_N_REQUESTS = 5000;
+
+        private readonly ILogger<CsharpLoadGenerator> _logger;
+        private readonly CsharpLoadGeneratorOptions _options;
+        private readonly string _cacheValue;
+
+        public CsharpLoadGenerator(ILoggerFactory loggerFactory, CsharpLoadGeneratorOptions options)
+        {
+            _logger = loggerFactory.CreateLogger<CsharpLoadGenerator>();
+
+            _options = options;
+
+            _cacheValue = new String('x', _options.cacheItemPayloadBytes);
+        }
+
+
+        public async Task Run()
+        {
+            string? authToken = System.Environment.GetEnvironmentVariable("MOMENTO_AUTH_TOKEN");
+            if (authToken == null)
+            {
+                throw new Exception("Missing required environment variable MOMENTO_AUTH_TOKEN");
+            }
+
+            var momento = new SimpleCacheClient(
+                authToken,
+                CACHE_ITEM_TTL_SECONDS,
+                _options.requestTimeoutMs
+            );
+
+            try
+            {
+                momento.CreateCache(CACHE_NAME);
+            }
+            catch (AlreadyExistsException)
+            {
+                _logger.LogInformation("cache '{0}' already exists", CACHE_NAME);
+            }
+
+
+            var numOperationsPerWorker = _options.totalNumberOfOperationsToExecute / _options.numberOfConcurrentRequests;
+            var totalNumRequestsExpected = _options.totalNumberOfOperationsToExecute * NUM_REQUESTS_PER_OPERATION;
+
+            var context = new CsharpLoadGeneratorContext();
+
+
+            var asyncResults = Enumerable.Range(0, _options.numberOfConcurrentRequests).Select<int, Task>(workerId =>
+                LaunchAndRunWorkers(
+                    momento,
+                    context,
+                    workerId + 1,
+                    numOperationsPerWorker
+                    )
+            );
+
+            var statsPrinterTask = LaunchStatsPrinterTask(context, totalNumRequestsExpected);
+
+            await Task.WhenAll(asyncResults);
+
+            await statsPrinterTask;
+            _logger.LogInformation("Done");
+        }
+
+
+        private async Task LaunchAndRunWorkers(
+            SimpleCacheClient client,
+            CsharpLoadGeneratorContext context,
+            int workerId,
+            int numOperations)
+        {
+            for (var i = 1; i <= numOperations; i++)
```

Comment:
do we want to add the concept of max concurrency like we had in other basic load generators? Not sure how un controlled concurrency effects latency in c# vs js or python but assuming still some latency impact on having no limit on this?

Reply:
Sorry missed `numberOfConcurrentRequests` below looks good 👍